src/blockdiag/tests/test_generate_diagram.py

```diff
@@ -62,12 +60,11 @@ def test_generate_with_separate():
     filtered = (f for f in files if re.search('separate', f))
     options = ['--separate']
 
-    for testcase in testcase_generator(basepath, mainfunc, filtered, options):
+    for testcase in _testcase_generator(basepath, mainfunc, filtered, options):
         yield testcase
```

Comment:
With nose, `yield` is used for test generator. AFAIK, pytest does not support it. So you need to rewrite whole of this method.

```
================== 238 passed, 2 xfailed, 3 warnings in 2.83s ==================
```
CI log for this PR.

```
Ran 710 tests in 45.516s
```
CI log for latest master: https://travis-ci.org/blockdiag/blockdiag/jobs/644720092?utm_medium=notification&utm_source=github_status